`integration-tests/grpc-external-proto-test` `generate-code` step fails to generate Java classes from proto files

### Describe the bug

Observed in the Camel Quarkus nightly build with Quarkus 999-SNAPSHOT. The project build fails at `integration-tests/grpc-external-proto-test` with:

```
Caused by: io.quarkus.bootstrap.prebuild.CodeGenException: Failed to generate Java classes from proto files: [/home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/src/main/proto/extended.proto, /home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/protoc-protos-from-dependencies/3a59f6ebb51a27a99eea361ddf0a831cb6314c7d/protobuf/base.proto, /home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/protoc-protos-from-dependencies/3a59f6ebb51a27a99eea361ddf0a831cb6314c7d/protobuf/role.proto, /home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/protoc-protos-from-dependencies/3a59f6ebb51a27a99eea361ddf0a831cb6314c7d/dir/proto-with-java_generic_services.proto, /home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/protoc-protos-from-dependencies/3a59f6ebb51a27a99eea361ddf0a831cb6314c7d/dir/my-proto.proto] to /home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/generated-sources/grpc with command /home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/com.google.protobuf-protoc-linux-x86_64-exe -I=/home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/protoc-protos-from-dependencies/3a59f6ebb51a27a99eea361ddf0a831cb6314c7d/dir -I=/home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/src/main/proto -I=/home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/protoc-protos-from-dependencies/3a59f6ebb51a27a99eea361ddf0a831cb6314c7d -I=/home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/protoc-protos-from-dependencies/3a59f6ebb51a27a99eea361ddf0a831cb6314c7d/protobuf -I=/home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/protoc-dependencies/6c3c225c5c070189dc57b09dcabf81b0f16d842d --plugin=protoc-gen-grpc=/home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/io.grpc-protoc-gen-grpc-java-linux-x86_64-exe --plugin=protoc-gen-q-grpc=/home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/quarkus-grpc2586788389285755291.sh --q-grpc_out=/home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/generated-sources/grpc --grpc_out=/home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/generated-sources/grpc --java_out=/home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/generated-sources/grpc /home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/src/main/proto/extended.proto /home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/protoc-protos-from-dependencies/3a59f6ebb51a27a99eea361ddf0a831cb6314c7d/protobuf/base.proto /home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/protoc-protos-from-dependencies/3a59f6ebb51a27a99eea361ddf0a831cb6314c7d/protobuf/role.proto /home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/protoc-protos-from-dependencies/3a59f6ebb51a27a99eea361ddf0a831cb6314c7d/dir/proto-with-java_generic_services.proto /home/runner/work/camel-quarkus/quarkus/integration-tests/grpc-external-proto-test/target/protoc-protos-from-dependencies/3a59f6ebb51a27a99eea361ddf0a831cb6314c7d/dir/my-proto.proto
     at io.quarkus.grpc.deployment.GrpcCodeGen.trigger (GrpcCodeGen.java:142)
     at io.quarkus.deployment.CodeGenerator.lambda$trigger$3 (CodeGenerator.java:189)
     at io.quarkus.deployment.CodeGenerator.callWithClassloader (CodeGenerator.java:164)
     at io.quarkus.deployment.CodeGenerator.trigger (CodeGenerator.java:186)
     at io.quarkus.deployment.CodeGenerator.initAndRun (CodeGenerator.java:76)
     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
     at java.lang.reflect.Method.invoke (Method.java:568)
     at io.quarkus.maven.GenerateCodeMojo.generateCode (GenerateCodeMojo.java:88)
     at io.quarkus.maven.GenerateCodeMojo.doExecute (GenerateCodeMojo.java:54)
     at io.quarkus.maven.QuarkusBootstrapMojo.execute (QuarkusBootstrapMojo.java:171)
```

I see the same thing locally.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

1. Clone `quarkusio/quarkus`
2. Checkout the `main` branch
3. `./mvnw clean install -Dquickly`

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/grpc/codegen/src/main/java/io/quarkus/grpc/deployment/GrpcCodeGen.java
+++ b/extensions/grpc/codegen/src/main/java/io/quarkus/grpc/deployment/GrpcCodeGen.java
@@ -14,6 +14,7 @@ import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.HashSet;
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Locale;
 import java.util.Set;
@@ -81,7 +82,7 @@ public class GrpcCodeGen implements CodeGenProvider {
         }
         Path outDir = context.outDir();
         Path workDir = context.workDir();
-        Set<String> protoDirs = new HashSet<>();
+        Set<String> protoDirs = new LinkedHashSet<>();
 
         try {
             List<String> protoFiles = new ArrayList<>();
@@ -108,6 +109,7 @@ public class GrpcCodeGen implements CodeGenProvider {
                     // proto file to the list of directories to include (it's a set, so no duplicate).
                     protoFiles.add(pathToProtoFile.toString());
                     protoDirs.add(pathToParentDir.toString());
+
                 }
             }
 
@@ -119,6 +121,7 @@ public class GrpcCodeGen implements CodeGenProvider {
 
                 List<String> command = new ArrayList<>();
                 command.add(executables.protoc.toString());
+
                 for (String protoDir : protoDirs) {
                     command.add(String.format("-I=%s", escapeWhitespace(protoDir)));
                 }
@@ -298,7 +301,9 @@ public class GrpcCodeGen implements CodeGenProvider {
                                         .normalize().toAbsolutePath();
                                 try {
                                     Files.createDirectories(protoUnzipDir);
-                                    protoDirectories.add(protoUnzipDir.toString());
+                                    if (filesToInclude.isEmpty()) {
+                                        protoDirectories.add(protoUnzipDir.toString());
+                                    }
                                 } catch (IOException e) {
                                     throw new GrpcCodeGenException("Failed to create directory: " + protoUnzipDir, e);
                                 }

